src/wallet/wallet.cpp

```diff
@@ -2483,6 +2483,10 @@ bool CWallet::SignTransaction(CMutableTransaction& tx, const std::map<COutPoint,
     // At this point, one input was not fully signed otherwise we would have exited already
     // Find that input and figure out what went wrong.
     for (unsigned int i = 0; i < tx.vin.size(); i++) {
+        if (input_errors.count(i)) {
+            // If there is already an error reported from the signing process, keep it
+            continue;
+        }
         // Get the prevout
```

Comment:
Looks like this block is unnecessary? The error `Input not found or already spent` is already set in `::SignTransaction`.

Reply:
And not sure why not set `Unable to sign input, missing keys` in `::SignTransaction`.